arm/Microsoft.Network/virtualNetworkGateways/deploy.bicep

```diff
@@ -182,6 +184,9 @@ var activeActive_var = (virtualNetworkGatewayType == 'ExpressRoute') ? bool('fal
 // Public IP variables
 var gatewayPipName1 = (length(gatewayPipName) == 0) ? '${virtualNetworkGatewayName}-pip1' : gatewayPipName[0]
 var gatewayPipName2 = activeActive_var ? ((length(gatewayPipName) == 1) ? '${virtualNetworkGatewayName}-pip2' : gatewayPipName[1]) : ''
+var varPublicIpZones = (empty(publicIpZones)) ? [
```

Comment:
Hola @rahalan , can you please cross reference this change with [this former PR](https://github.com/Azure/ResourceModules/pull/405) and discuss it with @torivara?

Reply:
Sounds to me like, instead of assuming a default, we should provide a parameter in the parameter file instead. However, I did not look into the details.

Reply:
@MrMCake there is a already a parameter for zones, but it defaults to none (which is ok). But if you use a gateway type that needs a zone, I think it makes sense to just default to zone 1, otherwise the deployment will fail. Alternatively we can say, it should fail and just change our parameter file to not have this scenario

Reply:
It still looks to me we're just undoing the exact change that was implemented in the previous PR as a fix. We're just 'hiding' it in a variable. Maybe we should take a look together.

Reply:
Why not do a conditional value here, @rahalan? The zone should be set only if the gateway is zone aware. There are several use cases where no zone is necessary, like dev/test. This should be an option.

Reply:
@torivara you are correct. I'll remove it. But will update the parameter file, as it will not deploy (zone aware VPN sku is set, but no PIP zone provided).

Reply:
The parameter file absolutely needs to be deployable, therefore the publicIpZones parameter must match the gatewaySku. I might have missed this earlier, when I updated the files.

Reply:
With the explicit parameter file we should be good.